languages/java/src/main/java/de/jplag/java/JavacAdapter.java

```diff
@@ -61,23 +68,18 @@ private Iterable<? extends CompilationUnitTree> executeCompilationTask(final Com
         return abstractSyntaxTrees;
     }
 
-    private String fileNameOf(File directory, final CompilationUnitTree ast) {
-        if (directory == null)
-            return ast.getSourceFile().getName();
-        else {
-            return Paths.get(directory.toURI()).relativize(Paths.get(ast.getSourceFile().toUri())).toString();
-        }
-    }
-
-    private int processErrors(Logger logger, DiagnosticCollector<Object> listener) {
-        int errors = 0;
+    private void processErrors(Logger logger, DiagnosticCollector<Object> listener) throws ParsingException {
         for (Diagnostic<?> diagnosticItem : listener.getDiagnostics()) {
             if (diagnosticItem.getKind() == javax.tools.Diagnostic.Kind.ERROR) {
+                File file = null;
+                if (diagnosticItem.getSource() instanceof JavaFileObject) {
+                    JavaFileObject fileObject = (JavaFileObject) diagnosticItem.getSource();
+                    file = new File(fileObject.toUri());
+                }
                 logger.error("{}", diagnosticItem);
-                errors++;
+                throw new ParsingException(file, diagnosticItem.getMessage(Locale.getDefault()));
```

Comment:
This is a bit of changed behavior, as you now only report the first error found by the compiler. Previously it would tell you, for example, all compiler errors it found. Maybe we add a constructor to the parsing exception that allows wrapping multiple issues (either multiple file/message pairs or just wrapping other parsing exceptions in a composite way).

Reply:
So you mean something like `ParsingException(Iterable<ParsingException> wrappedExceptions)`? Should be possible to do so.

Reply:
Added in https://github.com/jplag/JPlag/pull/688/commits/b72c4e505e09a45f7b8bb8b2450ef588e0245d2b